Library/Homebrew/test/utils/git_spec.rb

```diff
@@ -0,0 +1,175 @@
+    end
+  end
+
+  describe "#last_revision_of_file" do
+    it "returns last revision of file" do
+      expect(described_class.last_revision_of_file(repo, repo.to_s+"/"+file)).to eq("blah")
+    end
+
+    it "returns last revision of file based on before_commit" do
+      expect(described_class.last_revision_of_file(repo, repo.to_s+"/"+file, before_commit: "0..3")).to eq("brew")
+    end
+  end
+end
+
+describe Utils do
+  describe "::git_available?" do
+    it "returns true if git --version command succeeds" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(true)
+      expect(described_class.git_available?).to be_truthy
+    end
+
+    it "returns false if git --version command does not succeed" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(false)
+      expect(described_class.git_available?).to be_falsey
+    end
+
+    it "returns git version if already set" do
+      described_class.instance_variable_set(:@git, true)
+      expect(described_class.git_available?).to be_truthy
+      described_class.instance_variable_set(:@git, nil)
+    end
+  end
+
+  describe "::git_path" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns" do
+        expect(described_class.git_path).to eq(nil)
+      end
+    end
+
+    context "when git is available" do
+      before do
+        described_class.instance_variable_set(:@git, true)
+      end
+
+      it "returns path of git" do
+        expect(described_class.git_path).to eq("/Applications/Xcode.app/Contents/Developer/usr/bin/git")
+      end
+
+      it "returns git_path if already set" do
+        described_class.instance_variable_set(:@git_path, "git")
+        expect(described_class.git_path).to eq("git")
+        described_class.instance_variable_set(:@git_path, nil)
+      end
+    end
+  end
+
+  describe "::git_version" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns" do
+        expect(described_class.git_path).to eq(nil)
+      end
+    end
+
+    context "when git is available" do
+      before do
+        described_class.instance_variable_set(:@git, true)
+      end
+
+      it "returns version of git" do
+        expect(described_class.git_version).not_to be_nil
+      end
+
+      it "returns git_version if already set" do
+        described_class.instance_variable_set(:@git_version, "v1")
+        expect(described_class.git_version).to eq("v1")
+        described_class.instance_variable_set(:@git_version, nil)
+      end
+    end
+  end
+
+  describe "::git_remote_exists" do
+    context "when git is not available" do
+      before do
+        described_class.instance_variable_set(:@git, false)
+      end
+
+      it "returns true" do
+        expect(described_class.git_remote_exists("blah")).to be_truthy
```

Comment:
You're testing the right behaviour, but this one surprises me! I didn't realize this was the case.